test/cypress/integration/roles.spec.js

```diff
@@ -0,0 +1,47 @@
+import { generateRandomId } from '../support/utils';
+
+describe('Roles', () => {
+  const roleCode = generateRandomId();
+  beforeEach(() => {
+    cy.getOauth2Data();
+    cy.get('@oauth2Data').then((oauth2Data) => {
+      cy.keycloackLogin(oauth2Data, 'user');
+    });
+  });
+
+  afterEach(() => {
+    cy.get('@oauth2Data').then((oauth2Data) => {
+      cy.keycloackLogout(oauth2Data);
+    });
+    cy.clearCache();
+  });
+
+  describe('Add a new role', () => {
+    it('Should create a new role', () => {
+      cy.visit('/');
+      cy.contains('Users').click();
+      cy.contains('Roles').click();
+      cy.get('[data-cy=user-roles]').contains('Roles').should('be.visible');
+
+      cy.get('[data-cy=add-new-role]').click();
+      cy.get('[data-cy=add-user-role]').contains('Add').should('be.visible');
+      cy.get('[data-cy=add-role-name-input').type(roleCode);
+
+      cy.get('[data-cy=save-role').contains('Save').click();
+      cy.get('[data-cy=user-roles]').contains('Roles').should('be.visible');
+    });
+  });
+
+
+  describe('Edit an existing role', () => {
+    it('Should edit an existing role', () => {
+      cy.visit(`/role/edit/${roleCode}`);
```

Comment:
I wonder if we can think of adding this kind of thing in the support part,
Here is an example: https://github.com/jhipster/generator-jhipster/tree/main/generators/cypress/templates/src/test/javascript/cypress/support

Even if sometimes I find it "too much"